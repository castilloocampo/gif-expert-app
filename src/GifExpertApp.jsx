import { useState } from 'react';
import { PropTypes } from 'prop-types';
import AddCategory from './components/Category/AddCategory';
import GifGrid from './components/Gif/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

  // const categories = ["One Punch", "One Piece", "Dragon Ball"];
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory onSetCategories={setCategories} />
      <hr />
      <ol>
        {
          categories.map( category => (
            <GifGrid key={category} category={category} />
          ))
        }
      </ol>
    </>
  );
}

GifExpertApp.propTypes = {
  defaultCategories: PropTypes.array,
}

export default GifExpertApp;
