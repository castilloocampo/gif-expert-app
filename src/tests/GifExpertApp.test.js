
import { render, screen, waitFor } from '@testing-library/react';
import GifExpertApp from '../GifExpertApp';
describe('GifExpertApp', () => {
  const categories = [
    "One Piece",
    "Black Clover",
    "Attack on Titan"
  ];

  test('should show correctly', () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });

  test('should show a categories list', () => {
    const { container } = render(<GifExpertApp defaultCategories={categories}/>);
    expect(screen.getAllByRole("heading", { level: 3 }).length).toBe(categories.length);
    waitFor(() => {
      expect(container).toMatchSnapshot();
      expect(screen.getAllByTestId("gif-grid-item").length).toBe(categories.length*10);
    })
  });
});