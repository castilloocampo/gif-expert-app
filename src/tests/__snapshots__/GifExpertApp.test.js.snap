// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp should show a categories list 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <form
    aria-label="form"
  >
    <input
      placeholder="Ingrese búsqueda..."
      type="text"
      value=""
    />
  </form>
  <hr />
  <ol>
    <h3
      class="animate__animated animate__fadeIn"
    >
      One Piece
    </h3>
    <p
      class="animate__animated animate__flash"
    >
      Loading....
    </p>
    <div
      class="card-grid"
    />
    <h3
      class="animate__animated animate__fadeIn"
    >
      Black Clover
    </h3>
    <p
      class="animate__animated animate__flash"
    >
      Loading....
    </p>
    <div
      class="card-grid"
    />
    <h3
      class="animate__animated animate__fadeIn"
    >
      Attack on Titan
    </h3>
    <p
      class="animate__animated animate__flash"
    >
      Loading....
    </p>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;

exports[`GifExpertApp should show a categories list 2`] = `<div />`;

exports[`GifExpertApp should show correctly 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <form
    aria-label="form"
  >
    <input
      placeholder="Ingrese búsqueda..."
      type="text"
      value=""
    />
  </form>
  <hr />
  <ol />
</div>
`;
